### DISCUSSION TITLE
[Feature] Shared UI package between web and native

### DISCUSSION THREAD

**Discussion by tconroy** [t3-oss/create-t3-turbo#1031] [Ideas]
### Describe the feature you'd like to request

Right now, the `ui` package is only for `web`. However, given we are using both `TailWind` and `NativeWind`, it'd be great if we could have a single UI package for both platforms.

### Describe the solution you'd like to see

A single UI package for both Web and Native styling:
* share the same class configuration between web and native
* for using web: `import { Button } from '@acme/ui'`
* for using native: `import { NativeButton } from '@acme/ui'`

### Additional information

`react-native-web` might be a good way to normalize the API between web and native components.

**Comment by moritzWa** [comment]
I'm also interested in setting this up. Happy to explore this together. 

I found this guide: https://www.youtube.com/watch?v=TKvL2dasRrE 
and this relevant repo: https://github.com/gunnnnii/turbo-expo-trpc-starter by @gunnnnii

![image](https://github.com/t3-oss/create-t3-turbo/assets/42035131/e6533ada-89f7-490a-afab-e25c43800be0)

**Comment by moritzWa** [comment]
hmm, I noticed this in the readme now.

> No. Solito will not be included in this repo. It is a great tool if you want to share code between your Next.js and Expo app. However, the main purpose of this repo is not the integration between Next.js and Expo — it's the code splitting of your T3 App into a monorepo. The Expo app is just a bonus example of how you can utilize the monorepo with multiple apps but can just as well be any app such as Vite, Electron, etc.

> Integrating Solito into this repo isn't hard, and there are a few [official templates](https://github.com/nandorojo/solito/tree/master/example-monorepos) by the creators of Solito that you can use as a reference.

**Comment by tconroy** [comment]
I noticed that as well, @moritzWa. Personally, I think a shared UI package is different than something like Solito, which has the goal of leveraging NextJS for your native app (i.e: using the file based routing from your next app for your native app).

If this is out-of-scope for this project that is understandable, but I think something as simple as tailwind+nativeWind+react-native-web would enable a single UI package shared across both projects. But I'd love someone more knowledgeable than me to tell me why this isn't achievable! :-)

**Comment by shawnmclean** [comment]
I'm still not sure how solito works  unless its one app that ships to both mobile and web.

What I'm looking to do is write react native components in the shared UI package, more specifically [react-native-reusables](https://github.com/mrzachnugent/react-native-reusables) based on shadcn.

Then use that same component in both web and expo.

I'm stuck on getting the transpilers to work.

All I'm trying to do now is to render a react-native component inside a new next repo.

Has anyone seen anything like this? Is bringing in expo inside of next necessary? As seen in this adapter [here](https://github.com/expo/expo-webpack-integrations/tree/main/packages/next-adapter#readme).

**Reply by shawnmclean** [reply]
Well, I have something going on here: https://github.com/shawnmclean/sovoli

Its working on both nextjs and expo with some hacky stuff.

Not app components work yet.

**Comment by ofrades** [comment]
'use dom'

https://github.com/EvanBacon/expo-dom-components-canary-example
